packages/stencil/src/components/datacom-tooltip/datacom-tooltip.tsx

```diff
@@ -0,0 +1,287 @@
+   * @param element - Element to check is in viewport.
+   * @returns object with properties related element's position in viewport.
+   */
+  getViewportPositionData(element: HTMLElement) {
+    const { top, left, bottom, right } = element?.getBoundingClientRect();
+    const viewportHeight = window.innerHeight || document.documentElement.clientHeight;
+    const viewportWidth = window.innerWidth || document.documentElement.clientWidth;
+
+    const distanceFromTop = top;
+    const distanceFromLeft = left;
+    const distanceFromRight = viewportWidth - right;
+    const distanceFromBottom = viewportHeight - bottom;
+
+    return {
+      partiallyVisible: ((top > 0 && top < innerHeight) || (bottom > 0 && bottom < innerHeight)) && ((left > 0 && left < innerWidth) || (right > 0 && right < innerWidth)),
+      fullyVisible: top >= 0 && left >= 0 && bottom <= innerHeight && right <= innerWidth,
+      distanceFromBottom,
+      distanceFromLeft,
+      distanceFromRight,
+      distanceFromTop,
+      viewportWidth,
+      viewportHeight,
+    };
+  }
+
+  /**
+   * Logic for identifying appropriate position for tooltip
+   */
+  adjustTooltipPosition = () => {
+    const slottedElementPositionData = this.getViewportPositionData(this.slottedElement);
+    const verticalBuffer = 40;
+    const horizontalBuffer = 100;
+    const tooltipPositionBuffer = 10;
+
+    let mainPosition;
+    let subPosition;
+
+    // Check slotted element's distance from viewport edges to determine best cardinal direction for main tooltip position.
+    if (slottedElementPositionData.distanceFromBottom >= verticalBuffer) {
+      mainPosition = 'bottom';
+    } else if (slottedElementPositionData.distanceFromBottom < 0) {
+      mainPosition = 'top';
+    } else if (slottedElementPositionData.distanceFromLeft >= horizontalBuffer && slottedElementPositionData.distanceFromLeft > slottedElementPositionData.distanceFromRight) {
+      mainPosition = 'left';
+    } else if (slottedElementPositionData.distanceFromRight >= horizontalBuffer && slottedElementPositionData.distanceFromRight > slottedElementPositionData.distanceFromLeft) {
+      mainPosition = 'right';
+    } else {
+      mainPosition = 'top';
+    }
+
+    // Update tooltip position and verify tooltip position data for next steps.
+    this.updateTooltipPosition(mainPosition);
+    let tooltipPositionData = this.getViewportPositionData(this.tooltipElement);
+
+    // Now determine if sub position should be start, end or center depending on where element is placed in viewport.
+    if (mainPosition === 'top' || mainPosition === 'bottom') {
+      // Vertical cardinal directions
+      if (tooltipPositionData.distanceFromLeft < 0 && tooltipPositionData.distanceFromRight >= 0) {
+        subPosition = 'start';
+      } else if (tooltipPositionData.distanceFromLeft >= 0 && tooltipPositionData.distanceFromRight < 0) {
+        subPosition = 'end';
```

Comment:
Please can you test this again, might be needed to change the subPosition value "start" to "end" and "end" to "start".